src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java

```diff
@@ -204,6 +206,62 @@ public static String buildAppVersionTag(Run<?, ?> build) {
         return build.getParent().getDisplayName() + "-" + build.getNumber();
     }
 
+    /**
+     * Number of traces by severity
+     *
+     * @param traces
+     * @return String with the number of traces for each severity.
+     */
+    public static String getVulnerabilityInfoString(Traces traces) {
+        StringBuilder info = new StringBuilder();
+        int note = 0;
+        int low = 0;
+        int medium = 0;
+        int high = 0;
+        int critical = 0;
+
+        if (traces != null && traces.getTraces() != null && !traces.getTraces().isEmpty()) {
```

Comment:
This is a bit of a mess. "If traces is not null and traces get traces is not null and not traces is empty"

I get the checks here that you're being defensive and that's good. When things get complicated I like to spell it out explicitly with failing conditions.

```
if (traces == null || traces.getTraces() == null) {
  return info;
}

(go on with code)
```